providers/kaycloak_oidc.go

```diff
@@ -0,0 +1,20 @@
+package providers
+
+const keycloakOIDCProviderName = "Keycloak OIDC"
+
+// KeycloakOIDCProvider creates a Keycloak provider based on OIDCProvider
+type KeycloakOIDCProvider struct {
+	*OIDCProvider
+}
```

Comment:
So this is literally just the OIDC provider? What's the benefit of having this, where are we going?

Reply:
People are asking about the Roles from Access Token feature. I whipped this together as a starting point for them.

`EnrichSession` would likely be overridden to extract Roles from the Access Token claims like this PR started: https://github.com/oauth2-proxy/oauth2-proxy/pull/767/files

Reply:
https://github.com/oauth2-proxy/oauth2-proxy/issues/976#issuecomment-798931991

Reply:
I updated the dummy PR to show the others exactly where to plugin the role logic extension they wanted 👍